tools/@aws-cdk/generate-examples/lib/generate.ts

```diff
@@ -0,0 +1,381 @@
+  }
+  return exampleValue(context, param.type, param.name, 0);
+}
+
+/**
+ * Generate an example value of the given type.
+ */
+function exampleValue(context: ExampleContext, typeReference: reflect.TypeReference, name: string, level: number): Code {
+  // Process primitive types, base case
+  if (typeReference.primitive !== undefined) {
+    switch (typeReference.primitive) {
+      case spec.PrimitiveType.String: {
+        return new Code(`'${name}'`);
+      }
+      case spec.PrimitiveType.Number: {
+        return new Code('123');
+      }
+      case spec.PrimitiveType.Boolean: {
+        return new Code('false');
+      }
+      case spec.PrimitiveType.Any: {
+        // FIXME: add declaration with type any.
+        return new Code('anyValue', [new AnyAssumption('anyValue')]);
+      }
+      default: {
+        return new Code('---');
+      }
+    }
+  }
+
+  // Just pick the first type if it is a union type
+  if (typeReference.unionOfTypes !== undefined) {
+    // FIXME: which element should get picked?
+    const newType = getBaseUnionType(typeReference.unionOfTypes);
+    return exampleValue(context, newType, name, level);
+  }
+  // If its a collection create a collection of one element
+  if (typeReference.arrayOfType !== undefined) {
+    return Code.concatAll('[', exampleValue(context, typeReference.arrayOfType, name, level), ']');
+  }
+
+  if (typeReference.mapOfType !== undefined) {
+    return exampleValueForMap(context, typeReference.mapOfType, name, level);
+  }
+
+  if (typeReference.fqn) {
+    const fqn = typeReference.fqn;
+    // See if we have information on this type in the assembly
+    const newType = context.typeSystem.findFqn(fqn);
+
+    if (fqn in SPECIAL_TYPE_EXAMPLES) {
+      return new Code(SPECIAL_TYPE_EXAMPLES[fqn], [new Import(newType)]);
+    }
+
+    if (newType.isEnumType()) {
+      return new Code(`${newType.name}.${newType.members[0].name}`, [new Import(newType)]);
+    }
+
+    // If this is struct and we're not already rendering it (recursion breaker), expand
+    if (isStructType(newType) && !context.rendered.has(newType.fqn)) {
+      context.rendered.add(newType.fqn);
+      const ret = exampleValueForStruct(context, newType, level);
+      context.rendered.delete(newType.fqn);
+      return ret;
+    }
+
+    // For all other types  we will assume you already have a variable of the appropriate type.
+    return addAssumedVariableDeclaration(newType);
+  }
+
+  throw new Error('If this happens, then reflect.typeRefernce must have a new value');
+}
+
+function getBaseUnionType(types: reflect.TypeReference[]): reflect.TypeReference {
+  for (const newType of types) {
+    if (newType.fqn?.endsWith('.IResolvable')) {
+      continue;
+    }
+    return newType;
+  }
+  return types[0];
+}
+
+function addAssumedVariableDeclaration(type: reflect.Type): Code {
+  let newType = type;
+  if (type.isInterfaceType()) {
```

Comment:
I think we want to use this helper function that we have here: 

```ts
function isStructType(type: reflect.Type): type is reflect.InterfaceType {
  return type.isInterfaceType() && type.datatype;
}
```

However, this doesn't find any IXxx interfaces whereas just `type.isInterfaceType()` does.